app/src/main/java/com/example/bigowlapp/utils/PeriodicLocationCheckAlarmManager.java

```diff
@@ -0,0 +1,77 @@
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.SystemClock;
+
+import androidx.core.content.ContextCompat;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+
+// TODO: Might want to create a super AlarmManger given there are multiple now
+
+/**
+ * Create alarm that will force a location check periodically
+ */
+public class PeriodicLocationCheckAlarmManager {
+
+    private static final int REQUEST_CODE = 5;
+    private static final int REQUEST_CODE_CANCEL = 6;
+    private static final int NO_FLAG = 0;
+    private final Context context;
+    private final AlarmManager alarmManager;
+
+    public PeriodicLocationCheckAlarmManager(Context context) {
+        this.context = context;
+        this.alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+    }
+
+    public void setAlarm(long repeatIntervalMillis, long cancelTimeMillis) {
+        alarmManager.setRepeating(AlarmManager.ELAPSED_REALTIME, SystemClock.elapsedRealtime(),
+                repeatIntervalMillis, getPendingIntent());
+
+        alarmManager.set(AlarmManager.ELAPSED_REALTIME,
+                SystemClock.elapsedRealtime() + cancelTimeMillis,
+                getCancelPendingIntent());
+    }
+
+    public void cancelPeriodicLocationCheck() {
+        alarmManager.cancel(getPendingIntent());
+    }
+
+    private PendingIntent getPendingIntent() {
+        Intent intent = new Intent(context, PeriodicLocationCheckAlarmReceiver.class);
+        return PendingIntent.getBroadcast(context, REQUEST_CODE, intent, NO_FLAG);
+    }
+
+    private PendingIntent getCancelPendingIntent() {
+        Intent intent = new Intent(context, CancelPeriodicLocationCheckAlarmReceiver.class);
+        return PendingIntent.getBroadcast(context, REQUEST_CODE_CANCEL, intent, NO_FLAG);
+    }
+
+    public static class PeriodicLocationCheckAlarmReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            FusedLocationProviderClient locationClient = LocationServices.getFusedLocationProviderClient(context);
+            if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+                return;
+            }
+            locationClient.getCurrentLocation(LocationRequest.PRIORITY_HIGH_ACCURACY, null);
+        }
+    }
+
+    public static class CancelPeriodicLocationCheckAlarmReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            PeriodicLocationCheckAlarmManager locationCheckAlarmManager = new PeriodicLocationCheckAlarmManager(context);
+            locationCheckAlarmManager.cancelPeriodicLocationCheck();
+        }
+    }
+}
```

Comment:
Use an if statement instead of creating 2 different classes; it could be merged into 1 class.

Reply:
I'll be using the CancelPeriodicLocationCheckAlarmReceiver to make the missed schedule notification so it should be moved to a separate class on its own instead.

Reply:
I guess I'll make the inner class an outer class and rename it to reflect its more broad use then, although @ShawnGuerra-Bautista initial point was right if it was going to simply cancel the repeating alarm, which was what I made it for at first.